control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go

```diff
@@ -1225,6 +1225,12 @@ func (r *HostedControlPlaneReconciler) reconcileCloudProviderConfig(ctx context.
 		}); err != nil {
 			return fmt.Errorf("failed to reconcile Azure cloud config: %w", err)
 		}
+		withSecrets := manifests.AzureProviderConfigWithCredentials(hcp.Namespace)
```

Comment:
can we just drop the call to `ReconcileCloudConfig` and only keep `ReconcileCloudConfigWithCredentials`?

Reply:
No, we need the one without the credentials for the nodes and they are different besides just the creds themselves, the ones for the nodes needs `UseManagedIdentityExtension: true` and `UseInstanceMetadata: true` whereas the one for KAS/KCM needs those to be `false`

Reply:
make sense, thanks. Tiny addition to clarify that https://github.com/openshift/hypershift/pull/1061